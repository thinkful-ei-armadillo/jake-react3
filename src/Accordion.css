.Accordion{
  list-style-type: none;
  display: inline-flex;
}

li{
    width: 100px;
    text-align: center;
    margin: 50px;
    padding: 10px;
}

.Accordion button{
  
  width: 200px;
}

.Accordion p{
  
  padding: 10px;
  width: 100px;
  border-radius: 13px;
  background: rgba(170, 168, 168, 0.61);
  margin-left: 35px;
}