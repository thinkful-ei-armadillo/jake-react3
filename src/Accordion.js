import React, { Component } from 'react';

export default class Accorion extends Component {
  
  static defaultProps = {
  sections: [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ],
  }
  
  state = {
    lastClickedIndex: null
  }

  handleLastClicked = (lastIndex) => {
    this.setState({lastClickedIndex: lastIndex})

  }

  displayItem(section, index, lastClickedIndex) {
    return(
      <li className = "Item" key={index}>
      <button onClick={() => this.handleLastClicked(index)}>{section.title}</button>
      {(lastClickedIndex === index) && section.content}
      </li>
    )
  }

render() {
  const {lastClickedIndex} = this.state
  const {sections} = this.props
  return(
      <ul className = 'Accordion'>
      {sections.map((section, index) => 
        this.displayItem(section, index, lastClickedIndex)
      )}
      </ul>
  )
}
}