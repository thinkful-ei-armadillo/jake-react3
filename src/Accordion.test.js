import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Accordion from './Accordion';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const sectionsProp = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
]

describe(`Accordion Component`, () => {
  it('TThe component renders an empy li when the sections prop is not supplied', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('The component renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections = {sectionsProp}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('The component opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections = {sectionsProp}/>)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('The component only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections = {sectionsProp}/>)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
  
})