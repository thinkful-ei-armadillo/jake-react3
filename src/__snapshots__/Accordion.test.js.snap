// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component TThe component renders an empy li when the sections prop is not supplied 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Item"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Item"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Item"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component The component only opens the last section when multiple sections have been clicked 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Item"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Item"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Item"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
  </li>
</ul>
`;

exports[`Accordion Component The component opens a clicked section 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Item"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Item"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </li>
  <li
    className="Item"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion Component The component renders no sections as active by default 1`] = `
<ul
  className="Accordion"
>
  <li
    className="Item"
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    className="Item"
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    className="Item"
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
